### ISSUE TITLE
Rewarded Videos

### ISSUE THREAD

**Issue by avvie** [unity-plugins/Unity-Admob#15]
I saw from the readme and while coding that there are events for interstitial. Since i cant find documentation on the plugin. Is there an event or something for reward videos? I am using adcolony and sometimes it allows you to skip the reward video, and i need to check if that happened

**Comment by unity-plugins** [comment]
public static readonly string onAdLoaded = "onAdLoaded";
        public static readonly string onAdFailedToLoad = "onAdFailedToLoad";
        public static readonly string onAdOpened = "onAdOpened";
        public static readonly string adViewWillDismissScreen = "adViewWillDismissScreen";
        public static readonly string onAdClosed = "onAdClosed";
        public static readonly string onAdLeftApplication = "onAdLeftApplication";
        public static readonly string onRewardedVideoStarted = "onRewardedVideoStarted";
        public static readonly string onRewarded = "onRewarded";

**Comment by avvie** [comment]
thanks a lot

**Comment by avvie** [comment]
I am getting this with the events 
`
 I/Unity   (17467): NullReferenceException: Object reference not set to an instance of an object

I/Unity   (17467):   at admob.Admob+InnerAdmobListener.onAdmobEvent (System.String adtype, System.String eventName, System.String paramString) [0x00000] in <filename unknown>:0

I/Unity   (17467):   at admob.AdmobListenerProxy.onAdmobEvent (System.String adtype, System.String eventName, System.String paramString) [0x00000] in <filenameunknown>:0

I/Unity   (17467):   at System.Reflection.MonoMethod.Invoke (System.Object obj,BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) [0x00000] in <filename unknown>:0

I/Unity   (17467): Rethrow as TargetInvocationException: admob.AdmobListenerProxy.onAdmobEvent(System.String,System.String,System.String)
I/Unity   (17467):   at UnityEngine.AndroidJavaProxy.Invoke (System.String methodName, System.Object[] args) [0x00000] in <filename unknown>:0

I/Unity   (17467):   at UnityEngine.AndroidJavaProxy.Invoke (System.String methodName, UnityEngine.AndroidJavaObject[] javaArgs) [0x00000] in <filename unknown>:0

I/Unity   (17467):   at UnityEngine._And
`

Basically, calling adding the event and then starting the playback to the reward video doesnt work. Banner and normal interstitial work fine.

**Comment by unity-plugins** [comment]
updated

**Comment by avvie** [comment]
It worked for a couple of mins, then i got a couple of these in the log
` handler onAdmobEvent---onAdFailedToLoad   3_ERROR_CODE_NO_FILL
`
Is this a plugin thing? doesnt look like it, if it isnt you can close this thread. if you know how to deal with the no fill it would be greatly appreciated

Thanks a lot
